File: netlogo-gui/src/main/prim/etc/_readfromstring.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc;

import org.nlogo.core.CompilerException;
import org.nlogo.api.LogoException;
import org.nlogo.core.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.RuntimePrimitiveException;
import org.nlogo.nvm.Reporter;

public final class _readfromstring
    extends Reporter {


  @Override
  public Object report(final Context context)
      throws LogoException {
    try {
      return workspace.readFromString(argEvalString(context, 0));
    } catch (CompilerException error) {
      throw new RuntimePrimitiveException(context, this, error.getMessage());
    }
  }
}

File: netlogo-headless/src/main/prim/etc/_readfromstring.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.CompilerException
import org.nlogo.nvm.{ Context, Reporter }
import org.nlogo.nvm.RuntimePrimitiveException

class _readfromstring extends Reporter {
  override def report(context: Context): AnyRef =
    try workspace.readFromString(argEvalString(context, 0))
    catch {
      case e: CompilerException =>
        throw new RuntimePrimitiveException(context, this, e.getMessage)
    }
}
